src/__tests__/parseWithPointers.spec.ts

```diff
@@ -440,5 +444,14 @@ european-cities: &cities
         t: 4,
       });
     });
+
+    test('does not report duplicate errors for merged keys', () => {
+      const result = parseWithPointers(mergeKeysWithDuplicateProperties, {
+        mergeKeys: true,
+        ignoreDuplicateKeys: false,
+      });
+
+      expect(result.diagnostics).toEqual([]);
```

Comment:
@P0lip In order to protect this from potential future regressions, isn't there a way to also assess the expected result of the merges?

Reply:
Yeah, I'll add such an assertion for sure, before I mark it as ready for review, no worries.

Just in case you were concerned, we already have a few tests covering merge keys :)
https://github.com/stoplightio/yaml/blob/f8b22984d385867963d21e8bf63d21b4e3774b81/src/__tests__/parseWithPointers.spec.ts#L343
I just didn't cover diagnostics. :P

Reply:
@nulltoken just in case you were interested, a test case has been added.
Thanks for providing an excellent fixture I could use right away. :)

Reply:
>@nulltoken just in case you were interested, a test case has been added.

![image](https://user-images.githubusercontent.com/92363/64992299-e4cf4a00-d8d3-11e9-8907-d3d1edfcdf28.png)